src/signals/incident-management/components/CalendarInput/__tests__/CalendarInput.test.js

```diff
@@ -1,73 +1,77 @@
 
-    expect(
-      document.querySelectorAll('[class*=react-datepicker]').length
-    ).toBeGreaterThan(0);
+    expect(document.querySelectorAll('[class*=react-datepicker]').length).toBeGreaterThan(0);
   });
 
   it('renders a CustomInput component', () => {
-    const { getByTestId } = render(
-      withAppContext(<CalendarInput {...calendarInputProps} />)
-    );
+    render(withAppContext(<CalendarInput {...calendarInputProps} />));
 
-    expect(getByTestId('calendarCustomInputElement')).toBeInTheDocument();
+    expect(screen.getByTestId('calendarCustomInputElement')).toBeInTheDocument();
   });
 
-  it('renders a hidden input field', () => {
-    const { rerender } = render(
-      withAppContext(<CalendarInput {...calendarInputProps} />)
-    );
+  it('renders the selected date in the input field', () => {
+    const { rerender } = render(withAppContext(<CalendarInput {...calendarInputProps} />));
 
-    expect(
-      document.querySelector('input[type=hidden]')
-    ).not.toBeInTheDocument();
+    const element = screen.getByTestId('calendarCustomInputElement');
+    expect(element.querySelector('input').value).toEqual('');
 
     const selectedDate = new Date();
 
-    rerender(
-      withAppContext(
-        <CalendarInput {...calendarInputProps} selectedDate={selectedDate} />
-      )
-    );
+    rerender(withAppContext(<CalendarInput {...calendarInputProps} selectedDate={selectedDate} />));
 
-    expect(document.querySelector('input[type=hidden]')).toBeInTheDocument();
+    const elementWithDate = screen.getByTestId('calendarCustomInputElement');
+    expect(element.querySelector('input').value).toEqual(dateToString(selectedDate));
   });
 
   it('should call onSelect', () => {
     const onSelect = jest.fn();
-    const id = 'qux';
 
-    render(
-      withAppContext(
-        <CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />
-      )
-    );
+    render(withAppContext(<CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />));
 
     const inputElement = document.getElementById(id);
 
     expect(onSelect).not.toHaveBeenCalled();
 
     fireEvent.change(inputElement, { target: { value: '18-12-2018' } });
 
-    expect(onSelect).toHaveBeenCalledWith(
-      expect.any(Date),
-      expect.any(Object)
-    );
+    expect(onSelect).toHaveBeenCalledWith(expect.any(Date));
+  });
+
+  it('should focus on the input when a value is selected', () => {
+    const onSelect = jest.fn();
+    const selectedDate = new Date();
+
+    render(withAppContext(<CalendarInput {...calendarInputProps} id={id} onSelect={onSelect} />));
+
+    const inputElement = document.getElementById(id);
+    inputElement.focus();
+
+    expect(onSelect).not.toHaveBeenCalled();
+
+    fireEvent.click(inputElement);
+    fireEvent.keyDown(inputElement, { key: 'Enter', code: 13, keyCode: 13 });
+    fireEvent.keyDown(inputElement, { key: 'Enter', code: 13, keyCode: 13 });
```

Comment:
Try to use `userEvent`.
See my PR for using keypresses as well (update to the latest version of `user-event` needed).
https://github.com/Amsterdam/signals-frontend/pull/1310/files#diff-f0ea83e5cbf307979099162071be07dabf8ccd21fd36768cc81223f889f3060cR255-R257

Reply:
thanks! Will do in the further PR's